### PR TITLE
feat: some activity item row improvements in replay

### PR DESCRIPTION
**Pull request by pauldambra**
towards https://github.com/PostHog/posthog/issues/26098

* fixes a duration display
* adds an image tab when viewing an autocapture event in activity page and replay
* move flags to their own tab in event details
* add an elements tab for autocapture events in replay inspector

Screenshot 2024-11-29 at 20 39 52

Screenshot 2024-11-29 at 20 39 34

Screenshot 2024-11-29 at 20 49 01

### CODE REVIEW

File: frontend/src/scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent.tsx [hunk 1]
```diff
@@ -107,14 +107,20 @@ function StartedAt({ item }: { item: PerformanceEvent }): JSX.Element | null {
...
     }
-    if (duration === undefined) {
+    return duration ?? null
+}
+
+function DurationDescription({ item }: { item: PerformanceEvent }): JSX.Element | null {
+    const duration = durationMillisecondsFrom(item)
+    if (duration === null) {
         return null
     }
```
Comment (daibhin) [inline frontend/src/scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent.tsx:122]: Only ever a personal nit but I always find it easier to do this early return outside of a component. It makes things clearer elsewhere in the code that a component will be rendered. Also makes the component more flexible

File: frontend/src/lib/utils/event-property-utls.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return image.src ? image : null
+}
+
+export function AutocaptureImageTab({ elements }: { elements: ElementType[] }): JSX.Element | null {
+    const img = autocaptureToImage(elements)
+    if (img) {
+        return (
+            <div className="flex bg-bg-3000 items-center justify-center relative border-2">
+                {/* Transparent grid background */}
+                <div className="ImagePreview__background absolute h-full w-full" />
```
Comment (daibhin) [inline frontend/src/lib/utils/event-property-utls.tsx:23]: Not sure I love the transparent PNG style background but don't really have a suitable alternative. Did you consider just making it the default background color?

File: frontend/src/lib/taxonomy.tsx [hunk 3]
```diff
@@ -374,6 +374,16 @@ export const CORE_FILTER_DEFINITIONS_BY_GROUP = {
...
             examples: ['{"recordHeaders": false}'],
         },
+        $configured_session_timeout_ms: {
+            label: 'Configured session timeout',
+            description: <span>Configured session timeout in milliseconds.</span>,
+            examples: ['1800000'],
+        },
+        $replay_script_config: {
+            label: 'Replay script config',
+            description: <span>Sets an alternative recorder script for the web sdk.</span>,
```
Comment (daibhin) [inline frontend/src/lib/taxonomy.tsx:384]: Can these just be strings?

**Review by daibhin — APPROVED** [review]
Nothing blocking!

File: frontend/src/scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent.tsx:122]: in this case we're building a sentence and do want this to be null if there's no duration to render
and the sentence code would be super hard to parse imo

i remain torn on this pattern generally 🙈

File: frontend/src/lib/utils/event-property-utls.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/lib/utils/event-property-utls.tsx:23]: Yeah, it's because the first image I rendered this way had tall transparency and I though my render code was wrong. So I need to show the transparent part... but this is wrong... the transparent background should only be the bounds of the image so you can see where that is

Comment (pauldambra) [reply frontend/src/lib/utils/event-property-utls.tsx:26]: Screenshot 2024-12-02 at 12 04 31

Screenshot 2024-12-02 at 12 09 42

Screenshot 2024-12-02 at 12 10 19
Screenshot 2024-12-02 at 12 10 29

Comment (pauldambra) [reply frontend/src/lib/utils/event-property-utls.tsx:26]: so, we now show transparency _only_ where there is transparency
🙈
